### ISSUE TITLE
Document changes/tips required to embed an enketo form into an angular/ionic app

### ISSUE THREAD

**Issue by shankari** [enketo/enketo-core#623]
#### tl;dr ####

There are potentially nicer/more principled ways to do this, but this works, so I wanted to submit it in case it helps somebody else. I'm happy to check this in into some kind of doc repo, but couldn't find it so thought I would file it here to start with :)

1. build the enketo-core bundle using `grunt compile` from this repo
   1. you can check out the repo to do so, although I was able to build it from the npm installed version by calling `rollup` appropriately
   1. build
1. copy it over to `www/`. I put it in `www/manual_lib` to indicate that it was manually built.
1. edit it and add the line `window.FormModule = Form` to ensure that you can access the `Form` object
(https://github.com/e-mission/e-mission-docs/issues/376#issuecomment-488227423)

    ```
            return !!target;
        };

       window.FormModule = Form

        /**
         * Static method to obtain required enketo-transform version direct from class.
         */
    ```
1. Ensure that all your maps have the correct appearance tag
(https://github.com/e-mission/e-mission-docs/issues/376#issuecomment-488741737)
1. Copy paste the sample HTML directly into your `<ion-view>`. You do not want to use `<ion-header-bar>, <ion-footer-bar>` etc because jquery expects the elements to be under the same parent as the form.
1. Turn off touch handling for your newly added tags (https://github.com/e-mission/e-mission-docs/issues/376#issuecomment-488884563) - e.g.

    ```
     <article class="paper" data-tap-disabled="true">
    ```
1. Ensure that all the files you are loading remotely, including map tiles, are defined in the CSP. In particular, the default `config.js` loads not just tiles from OSM but also a client script from google maps. Either the config needs to be changed (stamen terrain is a great alternative to google satellite images) or the CSP needs to be changed.

####  Details

As I said in #619 I am embedding enketo forms into an angular1/ionic1 hybrid app.
The process was fairly straightforward and I did not need to make any changes except to the bundle creation.

angular uses its own module inclusion scheme, and it does not support ES6 style inclusions

```
import { Form } from 'enketo-core';
```
gives the error

```
Uncaught SyntaxError: Unexpected token {
```

while 

```
import Form from 'enketo-core';
```

gives the error 

```
Uncaught SyntaxError: Unexpected identifier
```

The obvious solution is to use some form of rollup/webpack/browserify, but the challenge is that I am not importing this into the main file of my app, but rather, into one of the angular modules. This is because surveying is only one part of more complex functionality. So running rollup on that module would not generate a whole-bundle app anyway.

It should be possible to run rollup on only that module's js file and then include the resulting bundle instead of the raw file, but I tried that with webpack and it didn't work. At this point, I had spent a day on javascript build issues, and so I decided to take the easy way out.

The bundle generated by rollup was almost perfect, except that because of the tree-shaking, it didn't actually export the form object for use in my module.

So although it was super-bad practice, I added one line to the generated bundle 

```
    window.FormModule = Form
```

to get it to work.

The other issues that I encountered were:
- need to specify appearance for maps (#620)
- need to change the CSP in `index.html` to include the hosts that we use. In particular, the default `config.js` loads not just tiles from but also a client script from google maps. Either the config needs to be changed (stamen terrain is a great alternative to google satellite images).

**Comment by MartijnR** [comment]
Thanks a lot for sharing! Will leave this out here for now. @magicznyleszek is working on a fancy documentation system for all Enketo repos (see PR in this repo) and I think this could be of use (at the very minimum as a link).

**Comment by shankari** [comment]
@MartijnR do you have a logo for "Powered by Enketo Core"?

We definitely want to give credit to your cool library, but would also like to highlight we are **embedding** the forms in our app, and in particular, the resulting data is not going to your servers.

The forms are embedded in an app that tracks location data long-term so people are a little sensitive about where their data goes.

**Comment by MartijnR** [comment]
@shankari, Certainly! You can use the one referenced in the source of the [README.md](https://raw.githubusercontent.com/enketo/enketo-core/master/README.md) at the bottom.

> We definitely want to give credit to your cool library, but would also like to highlight we are embedding the forms in our app, and in particular, the resulting data is not going to your servers.

Enketo actually doesn't manage any servers that store form data. So your use case is actually what we had enketo-core in mind for!

**Comment by shankari** [comment]
We already include the logo that you have at the bottom of the README. I wondered if there was a logo that said "Enketo Core" instead of Enketo. But I guess the website makes it clear that you support form building, so hopefully, users won't get too concerned.

**Comment by MartijnR** [comment]
Thanks! Forgot to mention, that I'd be happy to promote your app on the Enketo.org front page (free). Just send a logo and link or open an issue with those here:  https://github.com/enketo/enketo-website/issues

**Comment by shankari** [comment]
I haven't yet merged my PR to master - it turns out that the css changes from enketo causes the formatting of other parts of the app to break. Once @atton16 submits the PR for that change, and I merge to master, I'll be happy to be featured on your page.
